### ISSUE TITLE
[iOS] Low performance with 100+ markers with identical image

### ISSUE THREAD

**Issue by ericapply** [react-community/react-native-maps#1176]
### Description

I am using react-native-maps to draw about 100 markers on the map.
Each marker is using same Image

### Reproduction Steps and Sample Code

```
export default class Main extends Component {
  render() {
    return (
      <MapView style={StyleSheet.absoluteFill}>
        { coordinates.map(coordinate => {
              return (
                <Marker
                  key={coordinate.id}
                  coordinate={coordinate}
                  image={images.marker}
                />
              );
            })}
      </MapView>
    )
  }
}
```

### Possible Solution
Re-using UIImage
http://stackoverflow.com/questions/25555778/optimising-custom-marker-image-performance-with-google-maps-sdk-for-ios

Currently AIRMapMarker.m and AIRGoogleMapMarker.m does not re-use UIImage and constantly allocates a new one. This is a huge performance hit when you want to cluster markers and want to pan around (constant allocation and garbage collection of UIImage).

Currently there is a way to provide 'image' prop directly to Marker (the way I am doing in the code above). There should be a dictionary of <NSString, UIImage> for MapView and re-use the UIImage when imageSrc is same.

### Additional Information

* React Native version: ["0.41.2"]
* react-native-maps: ["^0.13.0"]
* Platform: [iOS]
* Development Operating System: [MacOS]
* Dev tools: [Xcode: 8.3, iPhone 7(10.3)]

![image](https://cloud.githubusercontent.com/assets/26330965/24479019/30f1a13e-1492-11e7-9f11-9dc2c24d9b14.png)

![image](https://cloud.githubusercontent.com/assets/26330965/24479005/26a5b166-1492-11e7-9636-3f3b5023beb0.png)

**Comment by hughvidos** [comment]
https://github.com/airbnb/react-native-maps/issues/1031

**Comment by hughvidos** [comment]
https://github.com/airbnb/react-native-maps/issues/1125

**Comment by ericapply** [comment]
@hughvidos I have found and confirmed that the lagging is due to the sheer number of UIViews being created for every marker regardless of rather they were using same image or not. I have tried re-using UIImage and 750+ markers gives me solid 55+ FPS even when panning around.

**Comment by ericapply** [comment]
https://github.com/airbnb/react-native-maps/pull/1187

**Comment by vinceyuan** [comment]
@ericapply Could you please take a look at this issue? https://github.com/airbnb/react-native-maps/issues/1031 Thanks

**Comment by ericapply** [comment]
@vinceyuan Can you try to use my PR and try to see if it still causes high CPU %?
    "react-native-maps": "ericapply/react-native-maps#d6e86de",
You must use following way in order to take advantage of my optimization
<MapView.Marker coordinate={coordinate} anchor={{x: 0.65, y: 1}} image={imageSrc}/>

**Comment by vinceyuan** [comment]
@ericapply I tried it. CPU % is still high. Because there is only one marker, reusing view does not help #1031
Anyway, thanks!

**Comment by mtt87** [comment]
Even with a few markers it's using a >75% of CPU.
I'm in the same situation where I need to use custom map style and I need to use Google Maps.

Did anyone find a workaround?

Thank you

**Comment by grundmanise** [comment]
Same here.. even with 1 marker `<MapView.Marker coordinate={coordinate} image={imageSrc}/>` the UI is noticeably lagging.

**Comment by juergengunz** [comment]
same for me, only laggy when using custom markers, even one marker makes it laggy

**Comment by krabbypattified** [comment]
+1

**Comment by alvelig** [comment]
Are you re-rendering map? Could someone provide a test repo reproducing performance problem?

**Comment by alvelig** [comment]
Follow in #369

**Comment by AngelVenchev** [comment]
Hey, I solved my problem and this could help somebody else too. I ended up using the image prop on the marker component to display whatever custom component I need. In my usecase I was implementing clustering using supercluster, I wanted to display a number of clusterized pins. I generated my images from here: https://dummyimage.com - about a hundred. Then a designer converted them to circles with a transparent background automatically with a Photoshop action and I loaded them in my application. 100 images were as small as 60kB so it's not a big hit that I have them all in the app.
